server/instance_cloud_oauth.go

```diff
@@ -80,11 +83,37 @@ func (ci *cloudOAuthInstance) GetClient(connection *Connection) (Client, error)
 
 func (ci *cloudOAuthInstance) getClientForConnection(connection *Connection) (*jira.Client, *http.Client, error) {
 	oauth2Conf := ci.GetOAuthConfig()
-
 	ctx := context.Background()
 	tokenSource := oauth2Conf.TokenSource(ctx, connection.OAuth2Token)
 	client := oauth2.NewClient(ctx, tokenSource)
 
+	// Get a new token if Access Token has expired
+	if ci.Plugin.IsAccessTokenExpired(connection.OAuth2Token) {
```

Comment:
The Golang OAuth library will automatically refresh the token based on the expiry, so this is an unnecessary check. Though we are now using this as a signal to store a new refresh token as well, which may work fine. To solve this in other plugins, we usually just call `TokenSource`, and see if the OAuth library performed a refresh of the _refresh_ token, and store the new refresh token if it changed from beforehand.